src/deadline/houdini_submitter/python/deadline_cloud_for_houdini/submitter.py

```diff
@@ -65,40 +65,74 @@ def _get_scene_asset_references(rop_node: hou.Node) -> AssetReferences:
         if os.path.isfile(path):
             asset_references.input_filenames.add(path)
 
-    rop_dir_map = {
-        # Mantra/karma
-        "Driver/ifd": "vm_picture",
-        "Driver/karma": "picture",
-        # Husk
-        "Lop/usdrender_rop": _husk_outputs,
-        # Arnold
-        "Driver/arnold": "ar_picture",
-        # Vray
-        "Driver/vray_renderer": "SettingsOutput_img_file_path",
-        "Sop/rop_vrayproxy": "filepath",
-        "Driver/rop_vrayproxy": "filepath",
-        # Renderman
-        "Driver/ris::3.0": _renderman_outputs,
-        # Redshift
-        "Driver/Redshift_ROP": "RS_outputFileNamePrefix",
-        # Geo
-        "Driver/geometry": "sopoutput",
-        "Driver/alembic": "filename",
-        "Sop/filecache": "file",
-    }
-    # collect output dirs for each ROP
-    all_inputs = rop_node.inputAncestors()
-    for node in all_inputs:
+    asset_references.output_directories.update(_get_output_directories(rop_node))
+
+    return asset_references
+
+
+def _get_output_directories(rop_node: hou.Node) -> set[str]:
+    """
+    Find and return the set of all output directories detected from the output
+    paths within the nodes leading into the Deadline Cloud ROP node
+    """
+
+    def find_output_dirs(node: hou.Node) -> set[str]:
+        """
+        Given a node, find and return the directories of the output path parms.
+        """
+        out_dirs = set()
         type_name = node.type().nameWithCategory()
         out_parm = rop_dir_map.get(type_name, None)
-        if out_parm is not None:
+
+        if out_parm:
             if callable(out_parm):
```

Comment:
alternative: you could also write this with inverted if-statements to de-indent and keep the relevant code to the conditions. Up to you though since this function is currently pretty small and it's not too hard to read.

ie.
```python
if not out_parm:
    return

if not callable(out_parm):
    path = node.parm(out_parm).eval()
    out_dirs.add(os.path.dirname(path))
    return out_dirs

if type_name == "Driver/wedge":
    ...
```